app.rb

```diff
@@ -72,14 +73,25 @@ def ifsc_data(code)
 end
 
 get '/:code' do
+  code = params['code']
   begin
-    data = ifsc_data(params['code'])
-    headers({
+    status = 404
+    data = ifsc_data(params['code']) || "Not found"
+
+    headers = {
       'Access-Control-Allow-Origin' => '*'
-    })
-    return json data if data
-    status 404
-    json "Not Found"
+    }
+
+
+    # Check if there is a redirect available
+    if settings.redirect_list.key? code
+      headers['Location'] = settings.redirect_list[code]
+      status = 302
```

Comment:
better to have redirect as permanent redirect `301`

Reply:
What if a client wants to fetch the new IFSC now, and the data for the old IFSC sometime later?